code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm

```diff
@@ -0,0 +1,193 @@
+			compiled_list[H.real_name] = H
+
+	if(LAZYLEN(compiled_list) == 0)
+		to_chat(user, "<span class='warning'>The items don't posses required fingerprints.</span>")
+		return
+
+	var/chosen_mob = input("Select the person you wish to curse","Your target") as null|anything in sortList(compiled_list, /proc/cmp_typepaths_asc)
+	if(!chosen_mob)
+		return ..()
+	curse(compiled_list[chosen_mob])
+	addtimer(CALLBACK(src, .proc/uncurse, compiled_list[chosen_mob]),timer)
+	. = ..()
+
+/datum/eldritch_knowledge/curse/proc/curse(mob/living/chosen_mob)
+	return
+
+/datum/eldritch_knowledge/curse/proc/uncurse(mob/living/chosen_mob)
+	if(!chosen_mob)
+		return
+
+/datum/eldritch_knowledge/summon
+	var/mob/living/mob_to_summon
+
+
+/datum/eldritch_knowledge/summon/on_finished_recipe(mob/living/user,list/atoms,loc)
+	. = ..()
+	var/mob/living/summoned = new mob_to_summon(loc)
+	message_admins("[summoned.name] is being summoned by [user.real_name] in [loc]")
+	var/list/mob/dead/observer/candidates = pollCandidatesForMob("Do you want to play as a [summoned.real_name]", ROLE_ECULTIST, null, ROLE_ECULTIST, 50,summoned)
+	if(!LAZYLEN(candidates))
+		return
+	var/mob/dead/observer/C = pick(candidates)
+	message_admins("[key_name_admin(C)] has taken control of ([key_name_admin(summoned)]).")
+	summoned.ghostize(0)
+	summoned.key = C.key
+
+	to_chat(summoned,"<span class='warning'>You are bound to [user.real_name]'s' will! Don't let your master die, protect him at all cost!</span>")
+///////////////
+///Base lore///
+///////////////
+
+/datum/eldritch_knowledge/spell/basic
+	name = "Break of dawn"
+	desc = "Starts your journey in the mansus. Allows you to transmute a soul bottle using a glass shard and a broken drinking bottle. Use a soul bottle on a dead person to harvest their soul. Use it then on ancient lore to  gain a charge. You can additionally harvest pierced realities. It takes a minute to harvest them and after you do so they are visible to everyone."
+	gain_text = "Gates of mansus open up to your mind."
+	next_knowledge = list(/datum/eldritch_knowledge/base_rust,/datum/eldritch_knowledge/base_ash,/datum/eldritch_knowledge/base_flesh)
+	cost = 0
+	spell_to_add = /obj/effect/proc_holder/spell/targeted/touch/mansus_grasp
+	required_atoms = list(/obj/item/living_heart)
+	route = "Start"
+
+/datum/eldritch_knowledge/spell/basic/recipe_snowflake_check(list/atoms, loc)
+	. = ..()
+	for(var/obj/item/living_heart/LH in atoms)
+		if(!LH.target)
+			var/datum/objective/A = new
+			LH.target = A.find_target()//easy way, i dont feel like copy pasting that entire block of code
+			qdel(A)
+			return FALSE
+		for(var/mob/living/carbon/human/H in atoms)
+			if(H.stat == DEAD)
+				return TRUE
+	return FALSE
+
+/datum/eldritch_knowledge/spell/basic/on_finished_recipe(mob/living/user, list/atoms, loc)
+	. = ..()
+	for(var/X in user.GetAllContents())
+		if(!istype(X,/obj/item/forbidden_book))
+			continue
+		var/obj/item/forbidden_book/FB = X
+		FB.charge++
```

Comment:
this adds a charge to ALL forbidden books on your person

Reply:
forgot a return lol